### ISSUE TITLE
typescript/consistent-type-imports: import used only as a `declare` class field's computed key is reported as type-only — typescript-eslint does not flag it, and the fix can break builds

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25805]
oxlint `1.78.0`, rule options `{"disallowTypeAnnotations": false, "fixStyle": "inline-type-imports"}`.

```ts
// keys.ts
export const StateKey = Symbol('state')

// example.ts
import {StateKey} from './keys'
import type {State} from './types'

export class Example {
	declare protected [StateKey]: State
}
```

oxlint reports on the `import {StateKey}` line:

> `error typescript(consistent-type-imports): All imports in the declaration are only used as types. Use 'import type'.`

typescript-eslint has never asked for this conversion, in either direction of its history:

| Variant | ESLint 7.23 + @typescript-eslint 6.21.0 | ESLint 9.39.5 + typescript-eslint 8.67.0 | oxlint 1.78.0 |
|---|---|---|---|
| `import {StateKey}` (above) | clean | clean | **error: convert to `import type`** |
| `import type {StateKey}` | **error: "Type import 'StateKey' is used by decorator metadata"** | clean | clean |
| `import {type StateKey}` | error (`no-import-type-side-effects`) | error (`no-import-type-side-effects`) | error (`no-import-type-side-effects`) |

Two consequences beyond the divergence itself:

1. **The suggested fix can break real pipelines.** `tsc` erases `declare` fields entirely, but at least one widely-used transform (rolldown-vite / Vite 8, i.e. Vitest 4's default pipeline) still emits the computed key of a `declare` field as a class static block under `useDefineForClassFields: false` — so after converting to `import type`, the emitted `static { StateKey; }` references an erased binding and every consumer crashes at class-definition time with `ReferenceError: StateKey is not defined`. That transform behavior is arguably its own bug, but since `tsc` itself accepts all three variants above, the lint conversion is the step that turns working code into broken code.
2. **With this option set the rule chases its own fixer.** `fixStyle: "inline-type-imports"` rewrites the import to `import {type StateKey}` — which `typescript/no-import-type-side-effects` (enabled alongside, as typescript-eslint recommends for this fixStyle) then errors on, demanding the top-level qualifier that consequence 1 shows is the dangerous form. The only shape both rules accept is the one such transforms can't run.

Expected: treat a reference in a `declare` member's computed property name as a value use (matching typescript-eslint 8), or at minimum stop auto-fixing it.
